src/mpyl/steps/deploy/k8s/helm.py

```diff
@@ -60,6 +71,33 @@ def __remove_existing_chart(
     )
 
 
+def install_with_values_yaml(
+    logger: Logger,
+    step_input: Input,
+    values: dict,
+    release_name: str,
+    chart_name: str,
+    namespace: str,
+    kube_context: str,
+) -> Output:
+    values_path = Path(step_input.project.target_path)
+    logger.info(f"Writing Helm values to {values_path}")
+
+    with open(values_path / Path("values.yaml"), mode="w+", encoding="utf-8") as file:
+        file.write(yaml.dump(values))
+
+    cmd = (
+        f"helm upgrade "
+        f"-i {release_name} "
+        f"-n {namespace} "
+        f'-f {values_path / Path("values.yaml")} '
+        f"--kube-context {kube_context} {chart_name}"
+    )
+    if step_input.dry_run:
+        return custom_check_output(logger, cmd + " --debug --dry-run")
+    return custom_check_output(logger, cmd)
```

Comment:
We could combine this part with the `install` function. Create a base function that has all the duplicated logic and pass the parts of the cmd that are different to append to the base cmd.

Reply:
done in 
34abdda